File: 01-week-1--starter-algorithms/02-day-3--fibonacci-series/javascript/fibonacci.js
function fibonacci(num) {
  if (num < 2) {return num}
  let fib = [0, 1]
  for (i = 0; i < num -1; ++i) {
    let total = fib[0] + fib[1]
    fib = [fib[1], total]
  }
  return fib[1]
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 0");
  console.log("=>", fibonacci(0));

  console.log("");

  console.log("Expecting: 1");
  console.log("=>", fibonacci(2));

  console.log("Expecting: 8");
  console.log("=>", fibonacci(6))

  console.log("Expecting: 55");
  console.log("=>", fibonacci(10));
}

module.exports = fibonacci;

// Please add your pseudocode to this file
  // if num is less than 2, return num (this accounts for num being 0 or 1)
  // initialize fib array with 0 and 1
  // create a for loop that iterates up to the value of num
  // create a total variable that stores the values of the fib array added together
  // reset the fib array with new values: the 2nd number from the previous version, and the value of total
  // return the second value from the fib array

File: 01-week-1--starter-algorithms/02-day-3--fibonacci-series/ruby/fibonacci.rb
def fibonacci(num)
  if num < 2
    num
  else
    fibonacci(num - 1) + fibonacci(num - 2)
  end
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 0"
  puts "=>", fibonacci(0)

  puts

  puts "Expecting: 1"
  puts "=>", fibonacci(2)

  puts "Expecting: 8"
  puts "=>", fibonacci(6)

  puts "Expecting: 55"
  puts "=>", fibonacci(10)

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
  # if num is less than 2, return num
  # else recursively return num - 1 + num - 2
# And a written explanation of your solution
  # I am using recursion to handle returning the correct value based on the number passed in, rather than storing a value and working forward from there.